examples/externalProfileDemo.html

```diff
@@ -0,0 +1,337 @@
+                    document.querySelector('.progress-label').innerHTML = errorMsg;
+                    document.querySelector('.progress-bar').className += ' progress-bar-danger';
+                    updateProgressBar(100);
+                }
+
+                /**
+                 * Parse the permissions array and check whether or not the match the specified required ones.
+                 *
+                 * @returns A boolean indicating if the user possesses the required permissions.
+                 */
+                function checkPermission(permissions, permissionValue) {
+                    let isAllowed = false;
+
+                    if (!permissions) {
+                        permissions = [];
+                    }
+
+                    if (permissionValue.match(/^[a-zA-Z0-9]+:\*$/)) {
+                        permissionValue = permissionValue.replace('*', '*:*');
+                    }
+
+                    const permissionsToValidate = permissionValue.split(':'),
+                        targetDomain = permissionsToValidate[0],
+                        targetEntity = permissionsToValidate[1],
+                        targetAction = permissionsToValidate[2];
+
+                    permissions.forEach(function (permission) {
+                        const permissions = permission.split(':'),
+                            domain = permissions[0],
+                            entity = permissions[1],
+                            actionSet = permissions[2];
+
+                        if (targetDomain === domain) {
+                            const matchesEntity = isPermission(targetEntity, entity),
+                                matchesAction = isPermission(targetAction, actionSet);
+
+                            if (matchesEntity && matchesAction) {
+                                isAllowed = true;
+                            }
+                        }
+                    });
+
+                    return isAllowed;
+                }
+
+                function isPermission(item, targetItem) {
+                    let isItem = item === '*' || targetItem === '*';
+                    if (!isItem) {
+                        isItem = item === targetItem;
+                    }
+                    return isItem;
+                }
+
+                /**
+                 * Determine the embedding PureCloud environment seeded on the query string or
+                 * being returned through the OAuth2 Implicit grant state hash param.
+                 *
+                 * @returns A string indicating the embedding PC env (e.g. mypurecloud.com, mypurecloud.jp); otherwise, null.
+                 */
+                function getEmbeddingPCEnv() {
+                    let result = null;
+
+                    if (window.location.hash && window.location.hash.indexOf('access_token') >= 0) {
+                        let oauthParams = extractParams(window.location.hash.substring(1));
```

Comment:
extractParams is actually pulled in from a utils.js in the other example that uses it.  You'll either want to copy that utils script and pull it in or we can look to make it a common set of utils.  Again, need to check the build to see if that's supported